internal/controllers/machine/machine_controller_noderef.go

```diff
@@ -87,6 +87,7 @@ func (r *Reconciler) reconcileNode(ctx context.Context, cluster *clusterv1.Clust
 		}
 		log.Error(err, "Failed to retrieve Node by ProviderID")
 		r.recorder.Event(machine, corev1.EventTypeWarning, "Failed to retrieve Node by ProviderID", err.Error())
+		conditions.MarkUnknown(machine, clusterv1.MachineNodeHealthyCondition, clusterv1.NodeConditionsFailedReason, "Failed to retrieve Node by ProviderID: "+err.Error())
```

Comment:
nit: fwiw we do `"Failed to get the node which is hosting this component: %v", err)` in other place, might want to align for consitency